mllib/src/main/scala/org/apache/spark/ml/feature/ChiSqSelector.scala

```diff
@@ -128,7 +130,8 @@ private[feature] trait ChiSqSelectorParams extends Params
   final val selectorType = new Param[String](this, "selectorType",
     "The selector type of the ChisqSelector. " +
       "Supported options: " + OldChiSqSelector.supportedSelectorTypes.mkString(", "),
-    ParamValidators.inArray[String](OldChiSqSelector.supportedSelectorTypes))
+    (value: String) => OldChiSqSelector.supportedSelectorTypes.map(_.toLowerCase(Locale.ROOT))
```

Comment:
Supported selector types should always be stored as lower case, please update corresponding code snippet in ```mllib.feature.ChiSqSelector``` from:
```
private[spark] val NumTopFeatures: String = "numTopFeatures"
......
```
to
```
private[spark] val NumTopFeatures: String = "numTopFeatures".toLowerCase
......
```